public/app/features/dashboard/components/PanelEditor/VisualizationTab.tsx

```diff
@@ -30,17 +30,20 @@ export const VisualizationTabUnconnected = React.forwardRef<HTMLInputElement, Pr
     const theme = useTheme();
     const styles = getStyles(theme);
 
-    const onPluginTypeChange = (meta: PanelPluginMeta) => {
-      if (meta.id === plugin.meta.id) {
-        onToggleOptionGroup(false);
-      } else {
-        changePanelPlugin(panel, meta.id);
-      }
-    };
+    const onPluginTypeChange = useCallback(
+      (meta: PanelPluginMeta) => {
+        if (meta.id === plugin.meta.id) {
+          onToggleOptionGroup(false);
+        } else {
+          changePanelPlugin(panel, meta.id);
+        }
+      },
+      [changePanelPlugin, onToggleOptionGroup, panel, plugin.meta.id]
+    );
 
     const onKeyPress = useCallback(
       (e: React.KeyboardEvent<HTMLInputElement>) => {
-        if (e.key === 'Enter') {
+        if (e.key === 'Enter' && plugin) {
```

Comment:
Nit: do we have incorrect typings on this component? Looks like plugin always should have a value.

Reply:
yep, fixed.